src/searchModal.js

```diff
@@ -0,0 +1,207 @@
+    instance.on('destroy', () => destroyModal());
+
+    /**
+     * Creates search modal, inits template selectors, and triggers initial search
+     */
+    function renderModal() {
+        initModal();
+        initUiSelectors();
+        searchButton.trigger('click');
+    }
+
+    /**
+     * Removes search modal
+     */
+    function destroyModal() {
+        instance.getElement()
+            .removeClass('modal')
+            .modal('destroy');
+        $('.modal-bg').remove();
+    }
+
+    /**
+     * Creates search modal
+     */
+    function initModal() {
+        instance
+        .getElement()
+        .addClass('modal')
+        .on('closed.modal', function() {
+            instance.destroy();
+        })
+        .modal({
+            disableEscape: true,
+            width: $( window ).width(),
+            minHeight: $( window ).height(),
+            modalCloseClass: 'modal-close-left'
+        })
+        .focus();
+    }
+
+    /**
+     * Inits template selectors and sets initial search query on search input
+     */
+    function initUiSelectors() {
+        searchButton = $('.btn-search', instance.getElement());
+        clearButton = $('.btn-clear', instance.getElement());
+        searchInput = $('.search-bar-container input', instance.getElement());
+        searchButton.on('click', search);
+        clearButton.on('click', clear);
+        searchInput.val(config.query);
+    }
+
+    /**
+     * Request search results and manages its results
+     */
+    function search() {
+        const query = searchInput.val();
+        if (query === '') {
+            clear();
+            return;
+        }
+        //throttle and control to prevent sending too many requests
+        const searchHandler = _.throttle(function searchHandlerThrottled(query){
+            if(running === false){
+                running = true;
+                $.ajax({
+                    url : config.url,
+                    type : 'POST',
+                    data :  {query : query},
+                    dataType : 'json'
+                }).done(buildSearchResultsDatatable).complete(function(){
+                    running = false;
+                });
+            }
+        }, 100);
+
+        searchHandler(query);
+    }
+
+    /**
+     * Creates a datatable with search results
+     * @param {object} data - search query results
+     */
+    function buildSearchResultsDatatable(data){
+        //update the section container
+        const $tableContainer = $('<div class="flex-container-full"></div>');
+        const section = $('.content-container', instance.getElement());
+
+        section.empty();
+        section.append($tableContainer);
+        $tableContainer.on('load.datatable', searchResultsLoaded);
+
+        //create datatable
+        $tableContainer.datatable({
+            url: data.url,
+            model : _.values(data.model),
+            emptyText: "No results were found.",
```

Comment:
Translation is missing

```suggestion
            emptyText: __("No results were found."),
```

Reply:
Instead I removed the property as the case for no results is managed outside the datatable, so the message will never be displayed.